File: src/main/java/pl/mjedynak/idea/plugins/builder/verifier/BuilderVerifier.java
package pl.mjedynak.idea.plugins.builder.verifier;

import com.intellij.psi.PsiClass;

public class BuilderVerifier {

    private BuilderVerifier() {
        throw new UnsupportedOperationException("Utility class");
    }

    private static final String SUFFIX = "Builder";

    public static boolean isBuilder(PsiClass psiClass) {
        String className = psiClass.getName();
        return className != null && className.endsWith(SUFFIX);
    }
}

File: src/main/kotlin/io/github/kingg22/kobra/builder/verifier/BuilderVerifier.kt
package io.github.kingg22.kobra.builder.verifier

import com.intellij.psi.PsiClass

public object BuilderVerifier {
    private const val SUFFIX = "Builder"

    @JvmStatic
    public fun isBuilder(psiClass: PsiClass): Boolean {
        val className = psiClass.name
        return className != null && className.endsWith(SUFFIX)
    }
}
